pkg/controller/infrastructure/templates/main.tpl.tf

```diff
@@ -4,12 +4,6 @@ provider "alicloud" {
   region = "{{ .alicloud.region }}"
 }
 
-// Import an existing public key to build a alicloud key pair
-resource "alicloud_key_pair" "publickey" {
-  key_pair_name = "{{ .clusterName }}-ssh-publickey"
-  public_key    = "{{ .sshPublicKey }}"
-}
-
```

Comment:
Okay, the key pair resource is removed from the terraform config but where is the key pair creation logic? Don't we still need to create an ECS key pair?

Reply:
We don't need a SSH key when we create a virtual machine. We rely on https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/gardeneruser/templates/scripts/reconcile.tpl.sh to set ssh-key

Reply:
Hm, I see https://github.com/gardener/gardenctl/pull/519. Is it correct that in order to have ssh working for gardenclt@v1 after this change, an operator should use version of gardenctl that includes https://github.com/gardener/gardenctl/pull/519? If this is the case we need to add a release not of type breaking for operators and we should ping for a new version of gardenctl that includes https://github.com/gardener/gardenctl/pull/519.

Reply:
I can confirm that gardenctl@v0.24.2 won't work with this change. gardener@master (hopefully gardenctl@0.25.0) will be required. So let's a release note of type breaking for operators that describes this case and let's ping the gardenctl folks for a release that includes gardener/gardenctl#519.

Reply:
Fix  gardener/gardenctl#519 is not target for ssh-key rotation. It fixed crash issue because of a regression. The  current master branch can ssh to both old and future(ssh key rotation) version of AliCloud extension.

Reply:
Yeah, this is correct. But currently latest gardenctl (0.24.2) won't work against provider extension that includes this PR.

Reply:
Hmm, I made a mistake. gardenctl has the regression in master not in v0.24.2.  I will release a note in this PR.

Reply:
I cannot follow what you mean by your latest comment. Yesterday I checked locally my statement 

> Yeah, this is correct. But currently latest gardenctl (0.24.2) won't work against provider extension that includes this PR. That why I wrote that first gardenctl@0.25.0 is required before considering this PR.

and made sure that it is the case. So I think we need gardenctl release that includes gardener/gardenctl#519, before considering this PR.

Reply:
I asked @neo-liang-sap  to  do a release.